module/VuFind/src/VuFind/Db/Service/ResourceService.php

```diff
@@ -47,21 +52,58 @@ class ResourceService extends AbstractService
     use \VuFind\Db\Service\ServiceAwareTrait;
     use LoggerAwareTrait;
 
+    /**
+     * Record loader
+     *
+     * @var Loader
+     */
+    protected $recordLoader;
+
+    /**
+     * Date converter
+     *
+     * @var DateConverter
+     */
+    protected $dateConverter;
+
+    /**
+     * Constructor
+     *
+     * @param EntityManager       $entityManager       Doctrine ORM entity manager
+     * @param EntityPluginManager $entityPluginManager VuFind entity plugin manager
+     * @param Loader              $loader              Record loader
+     * @param DateConverter       $converter           Date converter
+     */
+    public function __construct(
+        EntityManager $entityManager,
+        EntityPluginManager $entityPluginManager,
+        Loader $loader,
+        DateConverter $converter
+    ) {
+        parent::__construct($entityManager, $entityPluginManager);
+        $this->recordLoader = $loader;
+        $this->dateConverter = $converter;
+    }
+
     /**
      * Look up a row for the specified resource.
      *
-     * @param string $id     Record ID to look up
-     * @param string $source Source of record to look up
+     * @param string                            $id     Record ID to look up
+     * @param string                            $source Source of record to look up
+     * @param bool                              $create If true, create the row if
+     * it does not yet exist.
+     * @param \VuFind\RecordDriver\AbstractBase $driver A record driver for the
+     * resource being created (optional -- improves efficiency if provided, but will
+     * be auto-loaded as needed if left null).
      *
-     * @return Resource|null Matching row if found, null
+     * @return Resource|null Matching row if found or created, null
      * otherwise.
-     *
-     * Note that this method is partially migrated.
-     * Creation of a database record if does not yet exist has to be migrated.
      */
     public function findResource(
         $id,
-        $source = DEFAULT_SEARCH_BACKEND
+        $source = DEFAULT_SEARCH_BACKEND,
+        $create = true,
+        $driver = null
```

Comment:
Now that migration of this method is complete, can we delete the equivalent method from the Laminas\Db code?

Reply:
There are some many other instances where this method is used that have other dependencies, at this moment it's better to keep this method.

Reply:
Okay, that makes sense -- perhaps following this thread will reveal some other areas of refactoring that would be good next steps after this work is completed.